contracts/FIFOMarket.sol

```diff
@@ -39,6 +47,8 @@ contract FIFOMarket is
   uint256 private _noriFee;
   uint256 public priorityRestrictedThreshold;
   uint256 public totalSupply;
+  mapping(uint256 => mapping(address => Fee)) feeAddressToFeeIndex;
+  uint256 private _orderId;
```

Comment:
do we need to keep track of all this fee data in persistent storage? why don't we make these data structures in-memory only? (the orderid -> address -> Fee map).  then we wouldn't even need the concept of an order id because the data structures start fresh for every order and we can just have a `address -> Fee` map.  It occurred to me that it might be nice to have data about the orders available for the future but none of this is data that we can't get from looking at the `Certificate` contract and events, which can tell us how many orders there were (number of certs) AND way more information about the order such as who bought it, which actual removals went into it and for what amounts.  Since in the future we'll have a map from removalId -> noriFee, we'll be able to use that to reconstruct what the relative fee amounts were for the removals that comprised an order.  I think we can remove this from persistent storage, which should save hella gas since storage writes are so expensive.

Reply:
Can't declare mappings in memory in solidity :shrug:

Reply:
oh wow, very interesting! and a bummer.  explainer makes sense: https://ethereum.stackexchange.com/questions/25282/why-conceptually-cant-mappings-be-local-variables